test/event_manager_web/integrations/event_test.exs

```diff
@@ -65,25 +65,42 @@ defmodule EventManagerWeb.Schema.EventTest do
   end
 
   describe "query event" do
+    defp events_by_statuses(statuses) do
+      Enum.map(
+        statuses,
+        &%Event{
+          description: "Test #{&1}",
+          title: "test #{&1}",
+          location: "here",
+          status: &1,
+          start_time: DateTime.utc_now() |> DateTime.truncate(:second),
+          end_time: DateTime.utc_now() |> DateTime.truncate(:second)
+        }
+      )
+      |> Enum.map(&Event.changeset/1)
+      |> Enum.map(&EventManager.Repo.insert!/1)
+    end
+
     @query """
     query Event($id: ID!) {
       event(id: $id) { #{@event_data} }
     }
     """
 
-    test "responds to the event query" do
-      event = %Event{
-        description: "Test",
-        title: "test",
-        location: "here",
-        status: :draft,
-        start_time: DateTime.utc_now() |> DateTime.truncate(:second),
-        end_time: DateTime.utc_now() |> DateTime.truncate(:second)
-      }
+    @statuses ~w(draft published cancelled participations_closed)a
 
-      event = EventManager.Repo.insert!(event)
+    test "doesn't find a drafted event" do
+      [draft | _others] = events_by_statuses(@statuses)
 
-      {:ok, result} = Absinthe.run(@query, @schema, variables: %{"id" => event.id})
+      {:ok, result} = Absinthe.run(@query, @schema, variables: %{"id" => draft.id})
+
+      assert %{data: %{"event" => nil}} = result
+    end
+
+    test "finds a publish event" do
+      [_, published, _, _] = events_by_statuses(@statuses)
```

Comment:
Let's make `events_by_statuses(@statuses)` returning an hash like data structure like 
```
%{
 draft: %Event{ status: "draft" }
 # ...
}
```

Reply:
New commit coming with this fix!